scipy/fft/_pocketfft/setup.py

```diff
@@ -5,20 +5,28 @@ def pre_build_hook(build_ext, ext):
     cc = build_ext._cxx_compiler
     args = ext.extra_compile_args
 
-    std_flag = get_cxx_std_flag(build_ext._cxx_compiler)
+    std_flag = get_cxx_std_flag(cc)
     if std_flag is not None:
         args.append(std_flag)
 
     if cc.compiler_type == 'msvc':
         args.append('/EHsc')
     else:
-        try_add_flag(args, cc, '-fvisibility=hidden')
-
+        # Use pthreads if available
         has_pthreads = try_compile(cc, code='#include <pthread.h>\n'
                                    'int main(int argc, char **argv) {}')
         if has_pthreads:
             ext.define_macros.append(('POCKETFFT_PTHREADS', None))
-
+            if has_flag(cc, '-pthread'):
+                args.append('-pthread')
+                ext.extra_link_args.append('-pthread')
+            else:
+                from numpy.distutils import log
+                log.warn('Unknown compiler pthread flag')
```

Comment:
I'd make this a hard error instead. It is really cumbersome to build this and later find that it doesn't work. My current approach I'm testing is `ext.libraries.append('pthread')` (unconditionally)

The other is to try_link

            pthread_code = """#include <pthread.h>
                        void* thread_start(void* data){ return data; }
                        void dummy(){}
                        int main() {
                            pthread_t thread;
                            pthread_create(&thread, 0, thread_start, 0);
                            pthread_atfork(dummy, dummy, dummy);
                        }"""

and try `('', '-pthread', '-lpthread', '-lpthreads', '-lposix', '-lthread')` till one works